ipq806x: case-insensitive checks for qcom-smem partitions, required for kernel >=5.10

pkgadd opened the pull request:
qcom-smem traditionally displayed mtd partition names in upper case,
starting with kernel v5.10 it switches to normalizing the partition
names to lower case.

Adapt the sysupgrade support for the nbg6817 to check for both variants.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

**This patch is needed to go in before adding support for kernel v5.10 in https://github.com/openwrt/openwrt/pull/3954**, otherwise subsequent sysupgrades of the nbg6817 will fail, as the (then-) existing sysupgrade support code can't find the `DUAL_FLAG` partition.

Ansuel commented:
Also a similar change has already been merged for the luci-advanced-reboot app

pkgadd commented:
hmm, (at least-) these will need changing as well:
```
target/linux/ipq806x/base-files/etc/board.d/02_network: ucidef_set_interface_macaddr "lan" "$(mtd_get_mac_binary "0:ART" 0x6)"
target/linux/ipq806x/base-files/etc/board.d/02_network: ucidef_set_interface_macaddr "wan" "$(mtd_get_mac_binary "0:ART" 0x0)"
target/linux/ipq806x/base-files/etc/board.d/02_network: hw_mac_addr=$(mtd_get_mac_ascii 0:APPSBLENV ethaddr)
target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:               caldata_extract "0:ART" 0x1000 0x2f20
target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:               caldata_extract "0:ART" 0x1000 0x2f20
target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:               caldata_extract "0:ART" 0x1000 0x2f20
target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:               ath10k_patch_mac $(macaddr_add $(mtd_get_mac_ascii 0:APPSBLENV ethaddr) +1)
target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:               caldata_extract "0:ART" 0x5000 0x2f20
target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:               caldata_extract "0:ART" 0x5000 0x2f20
target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:               ath10k_patch_mac $(mtd_get_mac_ascii 0:APPSBLENV ethaddr)
target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:               caldata_extract "0:ART" 0x5000 0x2f20
target/linux/ipq806x/base-files/lib/upgrade/asrock.sh:  local mtdnum="$( find_mtd_index 0:BOOTCONFIG )"
```

adschm commented on target/linux/ipq806x/base-files/lib/upgrade/zyxel.sh:
```diff
@@ -90,7 +90,11 @@ zyxel_do_upgrade() {
+		# in upper case, since v5.10 they're normlized to lower case.
+		local dualflagmtd="$(find_mtd_part 0:dual_flag)"
+		[ -b $dualflagmtd ] || \
+			local dualflagmtd="$(find_mtd_part 0:DUAL_FLAG)"
```
Second "local" here should be removed.

adschm commented:
Commit title needs to be shortened.

pkgadd committed:
ipq806x: nbg6817: case-insensitive qcom-smem partitions

The out-of-tree qcom-smem patches traditionally displayed mtd partition names
in upper case, starting with the new mainline qcom-smem support in kernel v5.10,
it switches to normalizing the partition names to lower case.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

pkgadd committed:
ipq806x: g10: case-insensitive qcom-smem partitions

The out-of-tree qcom-smem patches traditionally displayed mtd partition names
in upper case, starting with the new mainline qcom-smem support in kernel v5.10,
it switches to normalizing the partition names to lower case.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

pkgadd committed:
ipq806x: ecw5410: case-insensitive qcom-smem partitions

The out-of-tree qcom-smem patches traditionally displayed mtd partition names
in upper case, starting with the new mainline qcom-smem support in kernel v5.10,
it switches to normalizing the partition names to lower case.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

pkgadd commented:
I've completed the changes necessary for the ZyXEL NBG6817, ASRock G10 and Edgecore ECW5410.

ASRock G10 and Edgecore ECW5410 have not been tested on the respective devices, nbg6817b only in its rebased form for kernel 5.10/ dsa.

pkgadd commented:
@Ansuel, @robimarko could you give this PR a casual sanity check, as I'm mostly guessing about the ASRock G10 and Edgecore ECW5410 (the results should be safe nevertheless, but better to confirm before removing the upper case fallbacks once kernel v5.4 gets dropped).

As-is, the patches are necessary to support kernel v5.4 and v5.10 in parallel

pkgadd commented:
Just to clarify, this PR needs to go in before kernel 5.10 support (regardless of swconfig or dsa) for ipq806x, otherwise subsequent sysupgrades are broken for the ZyXEL NBG6817 and ASRock G10. In addition to sysupgrade woes, firstboot conditions for ASRock G10, Edgecore ECW5410 and ZyXEL NBG6817 would fail to extract/ patch ART, set MAC addresses, etc.

robimarko commented:
It looks ok.
I would also suggest backporting the upstreamed SMEM parsers from kernel 5.12 to get rid of the outdated local one.

ynezz commented:
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git
